core/Memory_Model/Storage/TaskStorage.h

```diff
@@ -14,8 +14,7 @@ class TaskStorage{
  public:
   virtual const std::map<TaskID, std::shared_ptr<TaskEntity>>&    GetTasks() const = 0;
   virtual std::optional<std::shared_ptr<TaskEntity>>              GetTask(const TaskID& taskID) const = 0;
-  virtual std::optional<std::shared_ptr<TaskEntity>>              AddTask(const Task& task) = 0;
-  virtual std::optional<std::shared_ptr<TaskEntity>>              AddSubtask(const TaskID& rootTaskID, const Task& subtask) = 0;
+  virtual std::optional<std::shared_ptr<TaskEntity>>              AddTask(const TaskEntity& task) = 0;
```

Comment:
For what have you changed Task for TaskEntity?

Reply:
because for now, TaskEntity are creating in TaskRepository.

Reply:
Task was an input parameter, because ID for taskEntity was generated in taskStorage. And now it is in taskRepository.